gfx/drivers/vulkan.c

```diff
@@ -1810,8 +1818,14 @@ static bool vulkan_frame(void *data, const void *frame,
             vk->hw.src_queue_family, vk->context->graphics_queue_index);
    }
 
-   /* Upload texture */
-   if (frame && !vk->hw.enable)
+   /* Upload texture.
+    * If this was a mapped framebuffer previously handed off to the core as a
+    * software framebuffer, we don't update it, since we'd be reading from a
+    * GPU mapped pointer. This particular case happens when the core is paused
+    * and the menu is open - we're still rendering the previous frame.
+    */
+   if (frame && !vk->hw.enable &&
+      !vulkan_is_mapped_swapchain_texture_ptr(data, frame))
```

Comment:
Is it possible that frame refers to a different texture than the one we're rendering to, in which case it's doing staging -> staging copy?

Also, the GPU mapped pointer is CACHED, since cores might want to read from it (blending, etc).

Reply:
Anyways, at this point I don't know what's going on here, especially any magic state that lives outside the very core driver impl.